File: Trimestre 6/Proyecto final/API REST/src/models/detalleVenta.model.js
// models/detalleVenta.model.js
import { Model, DataTypes } from "sequelize";
import sequelize from "../config/connect.db.js";

class DetalleVenta extends Model {}

DetalleVenta.init(
  {
    idDetalleVenta: {
      type: DataTypes.INTEGER,
      autoIncrement: true,
      primaryKey: true,
    },
    Cantidad: {
      type: DataTypes.INTEGER,
      allowNull: false,
    },
    PrecioUnitario: {
      type: DataTypes.DECIMAL(10, 2),
      allowNull: false,
    },
    Subtotal: {
      type: DataTypes.DECIMAL(10, 2),
      allowNull: false,
    },
    Venta_id: {
      type: DataTypes.INTEGER,
      allowNull: false,
      // puedes agregar referencia si quieres
      // references: { model: "Venta", key: "idVenta" },
    },
    Producto_id: {
      type: DataTypes.INTEGER,
      allowNull: false,
      // references: { model: "Producto", key: "idProducto" },
    },
  },
  {
    sequelize,
    modelName: "DetalleVenta",
    tableName: "DetalleVenta",
    timestamps: false,
  }
);

export default DetalleVenta;

File: Trimestre 6/ArchivotablaFlutter/Flutter/frontend/lib/models/detalle_venta_model.dart
class DetalleVenta {
  int? idDetalleVenta;
  int Cantidad;
  double PrecioUnitario;
  double Subtotal;
  int Venta_id;
  int Producto_id;

  DetalleVenta({
    this.idDetalleVenta,
    required this.Cantidad,
    required this.PrecioUnitario,
    required this.Subtotal,
    required this.Venta_id,
    required this.Producto_id,
  });

  factory DetalleVenta.fromJson(Map<String, dynamic> json) {
    return DetalleVenta(
      idDetalleVenta: json["idDetalleVenta"],
      Cantidad: json["Cantidad"],
      PrecioUnitario:
          double.tryParse(json["PrecioUnitario"].toString()) ?? 0.0,
      Subtotal: double.tryParse(json["Subtotal"].toString()) ?? 0.0,
      Venta_id: json["Venta_id"],
      Producto_id: json["Producto_id"],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      "idDetalleVenta": idDetalleVenta,
      "Cantidad": Cantidad,
      "PrecioUnitario": PrecioUnitario,
      "Subtotal": Subtotal,
      "Venta_id": Venta_id,
      "Producto_id": Producto_id,
    };
  }
}
